### PR TITLE
Separate EM part from feature/machineStateManagerImprov,

### PR DESCRIPTION
**Pull request by rBekesh**
contains current implemetation of state confirm waiting mechanism

### CODE REVIEW

File: applications/ExecutionManager/src/execution_manager.cpp [hunk 1]
```diff
@@ -19,18 +19,21 @@ namespace {
   };
 } // anonymous namespace
 
-const string ExecutionManager::corePath =
-  string{"./bin/applications/"};
+const std::string ExecutionManager::corePath =
+  std::string{"./bin/applications/"};
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/execution_manager.cpp]: = "../bin/applications/"

File: applications/ExecutionManager/src/execution_manager.cpp [hunk 2]
```diff
@@ -157,62 +163,79 @@ void ExecutionManager::startApplication(const ProcessInfo& process)
...
                           + strerror(errno));
     }
-  } else {
+  } 
+  else
+  {
     // parent process
-    m_activeApplications.insert({process.processName, processId});
-  }
+    if (m_activeProcesses.count(process.processName))
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/execution_manager.cpp]: Is this branching needed? It seems that code does the same in both branches

File: applications/ExecutionManager/src/execution_manager.cpp [hunk 3]
```diff
@@ -157,62 +163,79 @@ void ExecutionManager::startApplication(const ProcessInfo& process)
...
-            << "\" for application with pid "
+  std::cout << "[ ExecutionManager ]:\t\tState \"" << applicationStateNames[static_cast<uint16_t>(state)]
+            << "\" for "
             << processId
-            << " received"
-            << " received"
+            << " received."
             << std::endl;
+
+  m_pidsWaitingForStateConfirm.erase(processId);
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/execution_manager.cpp]: Shouldn't be this structure secured with lock/mutex?

File: applications/ExecutionManager/src/execution_manager.cpp [hunk 4]
```diff
@@ -242,20 +265,42 @@ ExecutionManager::setMachineState(pid_t processId, string state)
...
+            << "\"" << state << "\"."
             << std::endl;
 
   return false;
 }
 
+void ExecutionManager::waitForStateConfirm()
+{
+  std::string listOfPid;
+  while (!m_pidsWaitingForStateConfirm.empty())
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/execution_manager.cpp]: How this happens that m_pidsWaitingForStateConfirm is updated in the background? I guess reportApplicationState is called in another state?

File: applications/ExecutionManager/src/execution_manager.cpp [hunk 5]
```diff
@@ -242,20 +265,42 @@ ExecutionManager::setMachineState(pid_t processId, string state)
...
+    for (auto pid : m_pidsWaitingForStateConfirm)
+    {
+      listOfPid += std::to_string(pid) + ", ";
+    }
+
+    std::cout << "[ ExecutionManager ]:\t\tWaiting confirm for application with pid == "
+              << listOfPid
+              << std::endl;
+
+    kj::NEVER_DONE.wait(kj::setupAsyncIo().waitScope);
```
Comment (adklimkiew) [inline applications/ExecutionManager/src/execution_manager.cpp]: What it does?

**Comment by rBekesh** [comment]
All functionality from this pull request was covered by changes on  feature/set-machine-state branch

**Pull request closed without merging**